src/WebJobs.Extensions.DocumentDB/Trigger/CosmosDBTriggerAttributeBindingProvider.cs

```diff
@@ -91,27 +92,55 @@ public Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext contex
 
         private string ResolveAttributeConnectionString(CosmosDBTriggerAttribute attribute)
         {
-            if (string.IsNullOrEmpty(_monitorConnectionString) &&
-                string.IsNullOrEmpty(attribute.ConnectionStringSetting))
+            string connectionString = this._monitorConnectionString;
+            if (!TryReadFromSettings(attribute.ConnectionStringSetting, connectionString, out connectionString))
+            {
+                throw new InvalidOperationException(
+                    string.Format(CultureInfo.CurrentCulture,
+                    "The CosmosDBTriggerAttribute.ConnectionStringSetting '{0}' property specified does not exist in the Application Settings.",
+                    attribute.ConnectionStringSetting));
+            }
+
+            if (string.IsNullOrEmpty(connectionString))
```

Comment:
I was trying to catch a certain scenario: If the Attribute is set but it's pointing to an un-existing setting. The more complete message is a bit generic, as a user, I would thing "But I did specify the Connection setting".